GoRogue.Debugger/Routines/RegionRoutine.cs

```diff
@@ -51,8 +60,11 @@ public void NextTimeUnit()
 
             // Rotate each original region by the new amount about its center, and add
             // it to the list of transformed regions
-            foreach (Region region in _originalRegions)
-                _transformedRegions.Add(region.Rotate(_rotation, region.Center));
+            foreach (RegionWithComponents region in _originalRegions)
+            {
+                var origin = region.GoRogueComponents.GetFirst<CenterOfRotation>().Origin;
```

Comment:
If `CenterOfRotation` is removed, this could be:
```suggestion
                var origin = region.GoRogueComponents.GetFirst<Point>("CenterOfRotation");
```

Reply:
Components added must be a reference type so I'm leaving this